lib/cli/src/commands/config.rs

```diff
@@ -3,38 +3,111 @@ use anyhow::{Context, Result};
+    #[clap(subcommand)]
+    Get(RetrievableConfigField),
+    /// Set a value in the current wasmer config
+    #[clap(subcommand)]
+    Set(StorableConfigField),
+}
+
+/// Value that can be queried from the wasmer config
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, clap::Subcommand)]
+pub enum RetrievableConfigField {
+    /// `prefix`
+    Prefix,
+    /// `bin-dir`
+    Bindir,
+    /// `includedir`
+    Includedir,
+    /// `libdir`
+    Libdir,
+    /// `libs`
+    Libs,
+    /// `cflags`
+    Cflags,
+    /// `pkg-config`
+    PkgConfig,
+    /// `registry.url`
+    #[clap(name = "registry.url")]
+    RegistryUrl,
+    /// `registry.token`
+    #[clap(name = "registry.token")]
+    RegistryToken,
+    /// `telemetry.enabled`
+    #[clap(name = "telemetry.enabled")]
+    TelemetryEnabled,
+    /// `update-notifications.url`
+    #[clap(name = "update-notifications.enabled")]
+    UpdateNotificationsEnabled,
+    /// `proxy.url`
+    #[clap(name = "proxy.url")]
+    ProxyUrl,
+}
 
-    /// Directory containing Wasmer executables.
-    #[clap(long, conflicts_with = "pkg-config")]
-    bindir: bool,
+/// Setting that can be stored in the wasmer config
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, clap::Subcommand)]
+pub enum StorableConfigField {
+    /// `registry.url`
+    #[clap(name = "registry.url")]
+    RegistryUrl(SetRegistryUrl),
+    /// `registry.token`
+    #[clap(name = "registry.token")]
+    RegistryToken(SetRegistryToken),
+    /// `telemetry.enabled`
+    #[clap(name = "telemetry.enabled")]
+    TelemetryEnabled(SetTelemetryEnabled),
+    /// `update-notifications.url`
+    #[clap(name = "update-notifications.enabled")]
+    UpdateNotificationsEnabled(SetUpdateNotificationsEnabled),
+    /// `proxy.url`
+    #[clap(name = "proxy.url")]
+    ProxyUrl(SetProxyUrl),
+}
 
-    /// Directory containing Wasmer headers.
-    #[clap(long, conflicts_with = "pkg-config")]
-    includedir: bool,
+/// Set the current active registry URL
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Parser)]
+pub struct SetRegistryUrl {
+    /// Url of the registry
+    #[clap(name = "URL")]
+    pub url: String,
+}
 
-    /// Directory containing Wasmer libraries.
-    #[clap(long, conflicts_with = "pkg-config")]
-    libdir: bool,
+/// Set or change the token for the current active registry
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Parser)]
+pub struct SetRegistryToken {
+    /// Token to set
+    #[clap(name = "TOKEN")]
+    pub token: String,
+}
 
-    /// Libraries needed to link against Wasmer components.
-    #[clap(long, conflicts_with = "pkg-config")]
-    libs: bool,
+/// Set if update notifications are enabled
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Parser)]
+pub struct SetUpdateNotificationsEnabled {
+    /// Whether to enable update notifications
+    #[clap(name = "ENABLED", possible_values = ["true", "false"])]
+    pub enabled: String,
```

Comment:
You can make this a bool, and use something like 

```
struct BoolString(bool);

impl std::str::FromStr for BoolString {
   ...
}
```

That way the other code becomes cleaner, and we can also accept something like `y` and `n` more easily.

Reply:
ok done